update geth dependency to version w/ v1.13.11 upstream commits

roberto-bayardo opened the pull request:
<!-- Contributions welcome! See https://github.com/ethereum-optimism/.github/blob/master/CONTRIBUTING.md -->

**Description**

updates the op-geth dependency to [v1.101311.0-rc.1](https://github.com/ethereum-optimism/op-geth/releases/tag/v1.101311.0-rc.1) which includes upstream geth commits up to v1.13.11.  The main changes required from this upgrade were around using uint256 instead of big.Int in various places.

Commit:
update geth dependency to version with upstream updates to v1.13.11

roberto-bayardo commented on op-e2e/e2eutils/wait/waits.go:
```diff
@@ -46,7 +46,7 @@ func ForReceipt(ctx context.Context, client *ethclient.Client, hash common.Hash,
 	for {
 		receipt, err := client.TransactionReceipt(ctx, hash)
-		if errors.Is(err, ethereum.NotFound) {
+		if errors.Is(err, ethereum.NotFound) || (err != nil && err.Error() == "transaction indexing is in progress") {
```
this is ugly but unfortunately the [underlying error](https://github.com/ethereum/go-ethereum/blob/7bb3fb1481acbffd91afe19f802c29b1ae6ea60c/internal/ethapi/errors.go#L61) isn't exposed by geth (it's in /internal). I suppose we could expose it in op-geth though if we don't mind more diffs.

trianglesphere replied:
Could we do a string.Contains to ensure this doesn't fail if the error gets wrapped?

roberto-bayardo replied:
good call, done.

trianglesphere commented:
@roberto-bayardo it looks like you also need some op-program updates for the new uint256 type

Commit:
fix op-program tests

roberto-bayardo commented:
> @roberto-bayardo it looks like you also need some op-program updates for the new uint256 type

thanks, fixed.  Also updated the tests to use the monorepo blob library instead of the crate-crypto one.

Commit:
update op-geth dep to tagged commit

maurelian reviewed (approved):
Approving changes in contracts package
